app/controllers/application_modal_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ApplicationModalController < ApplicationController
```

Comment:
Not necessary. But, it seems to me, that creating a new controller is a bit cleaner solution then placing all the code inside the ApplicationController.